pkg/trace/agent/agent.go

```diff
@@ -56,6 +56,9 @@ type Agent struct {
 	obfuscator     *obfuscate.Obfuscator
 	cardObfuscator *ccObfuscator
 
+	// SpanProcessor is used to apply extra logic on span once they have been received
+	SpanProcessor traceutil.SpanProcessor
```

Comment:
```suggestion
	ProcessSpan func(*pb.Span)
```
Why do we even need any interface?

Also, I think you can drop `tags`. You have access to it in the `serverless/trace` package via `s.ta.conf.GlobalTags`. Right?

Reply:
You're right, a function is enough. About the tags, we can't access `conf` from `t.sa` from the serverless package as it's not exported but I think it will indeed be more re-usable if we have just one param : the span. So I've found another way to get those tags.